x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/BaseSearchableSnapshotIndexInput.java

```diff
@@ -98,4 +85,10 @@ private void ensureValidPosition(long position) {
             throw new IllegalArgumentException("Position [" + position + "] is invalid");
         }
     }
+
+    boolean assertHashIsNotEqualToContent() {
```

Comment:
I think it'd be better to assert this (inline) in the constructor.

Reply:
Good point